File: app/src/main/java/com/digicoffer/lauditor/CommonFiles/GlobalFiles/CalendarHelper.java
package com.digicoffer.lauditor.CommonFiles.GlobalFiles;

import android.content.Context;
import android.graphics.drawable.Drawable;
import com.applandeo.materialcalendarview.CalendarUtils;
import com.applandeo.materialcalendarview.utils.DayColorsUtils;

public class CalendarHelper {
    public static Drawable getDrawableText(Context context, String string, int color, int size) {
        return CalendarUtils.getDrawableText(context, string, null, color, size);
    }

    public static int parseColor(Context context, int colorRes) {
        return DayColorsUtils.parseColor(context, colorRes);
    }
}

File: app/src/main/java/com/digicoffer/lauditor/CommonFiles/GlobalFiles/CalendarHelper.kt
package com.digicoffer.lauditor.CommonFiles.GlobalFiles

import android.content.Context
import android.graphics.drawable.Drawable

class CalendarHelper {
    companion object {
        @JvmStatic
        fun getDrawableText(context: Context, string: String, color: Int, size: Int): Drawable? {
            return try {
                val clazz = Class.forName("com.applandeo.materialcalendarview.CalendarUtils")
                val method = clazz.getMethod(
                    "getDrawableText",
                    Context::class.java,
                    String::class.java,
                    android.graphics.Typeface::class.java,
                    Int::class.javaPrimitiveType ?: Int::class.java,
                    Int::class.javaPrimitiveType ?: Int::class.java
                )
                method.invoke(null, context, string, null, color, size) as? Drawable
            } catch (e: Exception) {
                e.printStackTrace()
                null
            }
        }

        @JvmStatic
        fun parseColor(context: Context, colorRes: Int): Int {
            return try {
                val clazz = Class.forName("com.applandeo.materialcalendarview.utils.DayColorsUtils")
                val method = clazz.getMethod(
                    "parseColor",
                    Context::class.java,
                    Int::class.javaPrimitiveType ?: Int::class.java
                )
                method.invoke(null, context, colorRes) as Int
            } catch (e: Exception) {
                e.printStackTrace()
                0
            }
        }
    }
}
